TACC/Core-Styles
# feat(components): form patterns (c-form, django.cms.forms)

## Overview

Create "C Form" and "Django CMS Form" **prototype** patterns.

## Related

- [TUP-369](https://jira.tacc.utexas.edu/browse/TUP-369)
- [FP-1178](https://jira.tacc.utexas.edu/browse/FP-1178)
- required by https://github.com/TACC/Core-CMS/pull/___
- required by https://github.com/TACC/tup-ui/pull/___

## Changes

- changed demo to load new CMS styles
- added `c-form` pattern
- added `django-cms-forms` pattern
    <sup>(temporary for testing)*</sup>
- changed `s-affixed-input-wrapper` to work with `c-form`

<sub>* This pattern belongs in CMS.</sub>

## Testing

0. Build and serve demo.
1. Open and review "Components" → "C Form" pattern.
2. Open and review "Components" → "Django CMS Form" pattern.
3. Expect good markup to work with.

## UI

<details><summary>C Form</summary>

![C Form](https://user-images.githubusercontent.com/62723358/205183905-e48e2870-57e1-4ed6-87e6-f02ae1c339f3.png)

</details>
<details><summary>Django CMS Forms</summary>

![Django CMS Forms 1](https://user-images.githubusercontent.com/62723358/205183978-0e901d39-8d1f-4be0-8e77-e9cd7a303546.png)
![Django CMS Forms 2](https://user-images.githubusercontent.com/62723358/205183981-987c12d0-28d1-4396-b573-05ec56cc6b07.png)
![Django CMS Forms 3](https://user-images.githubusercontent.com/62723358/205183983-40b389e4-0bb1-46d7-89b7-793cc2756b21.png)

</details>

## Notes

- This is just a scaffold to work from in a follow-up PR.
- I need to finish #78 before I can develop these reliably.
